test/TestCaseReader.cpp

```diff
@@ -137,6 +144,54 @@ pair<SourceMap, size_t> TestCaseReader::parseSourcesAndSettingsWithLineNumber(is
 				if (sources.count(currentSourceName))
 					BOOST_THROW_EXCEPTION(runtime_error("Multiple definitions of test source \"" + currentSourceName + "\"."));
 			}
+			else if (boost::algorithm::starts_with(line, externalSourceDelimiterStart) && boost::algorithm::ends_with(line, sourceDelimiterEnd))
+			{
+				string externalSourcePath = boost::trim_copy(line.substr(
+					externalSourceDelimiterStart.size(),
+					line.size() - sourceDelimiterEnd.size() - externalSourceDelimiterStart.size()
+				));
+				string externalSourceName = externalSourcePath;
+
+				// Does the external source define a remapping?
+				size_t remappingPos = externalSourcePath.find('=');
+				if (remappingPos != string::npos)
+				{
+					externalSourceName = boost::trim_copy(externalSourcePath.substr(0, remappingPos));
+					externalSourcePath = boost::trim_copy(externalSourcePath.substr(remappingPos + 1));
+				}
+
+				string externalSourceContent;
+				fs::path parent_path(m_filename);
+				parent_path = parent_path.normalize().remove_filename();
+
+				fs::path path = parent_path / fs::path(externalSourcePath);
+
+				if (fs::exists(path))
+					externalSourceContent = util::readFileAsString(path.string());
+				else
+					BOOST_THROW_EXCEPTION(runtime_error(string("External Source '"+ externalSourcePath +"' not found.")));
+
+				if (!externalSourceName.empty())
+				{
+					ErrorList errorList;
+					ErrorReporter errorReporter(errorList);
+					Parser parser(errorReporter, langutil::EVMVersion{});
+					shared_ptr<Scanner> scanner = make_shared<Scanner>(CharStream(externalSourceContent, externalSourceName));
+					ASTPointer<SourceUnit> sourceUnit = parser.parse(scanner);
+					if (sourceUnit)
+						for (auto const& node: sourceUnit->nodes())
+							if (auto import = dynamic_cast<ImportDirective const*>(node.get()))
+							{
+								fs::path parent(externalSourcePath);
+								parent = parent.parent_path();
```

Comment:
Please give it a more self-explanatory name. There are multiple parent dirs in this function.
```suggestion
								fs::path externalSourceParentDir = fs::path(externalSourcePath).parent_path();
```